### PR TITLE
Leaf/configure axios

### PR DESCRIPTION
**Pull request by owwix**
- Axios configured. Front end can now communicate with django server. 
- CSRF middleware temporarily disabled, this will be reenabled once we properly implement CSRF tokens.

### CODE REVIEW

File: meshapp/src/config/axiosConfig.ts
Comment (rezatnoMsirhC) [inline meshapp/src/config/axiosConfig.ts:1]: I could see us needing to expand this from a "config" to a class in order to support and organize MeshApp-specific logic. Have you considered writing a wrapper-class here instead? For example, maybe we want to support a default error-handler for POST-requests without having to write out the try-catch every time.

File: meshapp/src/config/axiosConfig.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { axiosInstance } from './config/axiosConfig'
+
+Then you can make calls as such -> axiosInstance.get(), axiosInstance.post(), etc
+*/
+
+const baseURL: string = 'http://localhost:8000/'
+
+export const axiosInstance = axios.create({
+    baseURL: baseURL,
+  });
```
Comment (rezatnoMsirhC) [inline meshapp/src/config/axiosConfig.ts:18]: How about adding a default-timeout here?

**Review by rezatnoMsirhC — CHANGES_REQUESTED** [review]

File: meshapp/src/config/axiosConfig.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+/*
+Base URL is configured such that when you are using Axios to contact django endpoint -> 'http://localhost:8001/user/reset'
+You will need pass this request endpoint into axios call -> 'user/reset'
+
+To use axios with your form, import instance as such: 
+
+import { axiosInstance } from './config/axiosConfig'
+
+Then you can make calls as such -> axiosInstance.get(), axiosInstance.post(), etc
```
[the remark this answers is not in the document]
Comment (owwix) [reply meshapp/src/config/axiosConfig.ts:11]: done 👍

File: meshapp/src/config/axiosConfig.ts (hunk 1 above)
Comment (owwix) [reply meshapp/src/config/axiosConfig.ts:21]: done 👍

File: meshapp/src/config/axiosConfig.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
```
[the remark this answers is not in the document]
Comment (owwix) [reply meshapp/src/config/axiosConfig.ts:1]: > How is this working without `axios` in [package.json](https://github.com/LetsMesh/Site/blob/leaf/configure_axios/meshapp/package.json#L5-L22)?

just fixed, pushed my package.json changes.

File: meshapp/src/config/axiosConfig.ts
Comment (owwix) [reply meshapp/src/config/axiosConfig.ts:1]: Created an issue for this: https://github.com/LetsMesh/Site/issues/211

**Review by rezatnoMsirhC — APPROVED** [review]

**Review by Darren52BT — APPROVED** [review]

**Pull request merged**